crates/bevy_reflect/bevy_reflect_derive/src/lib.rs

```diff
@@ -175,5 +219,23 @@ pub fn impl_from_reflect_value(input: TokenStream) -> TokenStream {
         &def.type_name,
         &def.generics,
         def.traits.unwrap_or_default(),
+        None,
+    ))
+}
+
+#[proc_macro]
+pub fn impl_type_name(input: TokenStream) -> TokenStream {
```

Comment:
Nit: Could you add some quick documentation here? Basically stating that this is mainly for internal use and specifically for types we cannot place a derive on.